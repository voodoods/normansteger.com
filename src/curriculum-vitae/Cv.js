import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Navbar from '../navbar/Navbar'
import JobCard from './job-card/JobCard'
import StyledCv from './Cv.styled'
import ContentNotFound from '../content-not-found/ContentNotFound'

import { jobSet } from './job-set'

class Cv extends Component {
  constructor(props) {
    super(props)

    this.state = {
      jobSet,
    }
  }

  getGroup(groupName) {
    return jobSet.filter(job => job.group === groupName)
  }

  render() {
    const internships = this.getGroup('Internships')
    const training = this.getGroup('Training')
    const employments = this.getGroup('Employments')
    const certificates = this.getGroup('Certificates')

    return (
      <StyledCv className="Cv pure-u-lg-3-5 pure-u-md-1 pure-u-sm-1">
        {this.state.jobSet.length === 0 && <ContentNotFound />}

        <Navbar activePath={this.props.location.pathname} />

        {this.state.jobSet.length > 0 && (
          <div>
            <h3>
              <a name="employments">Employments</a>
            </h3>
            {employments.map((job, index) => {
              return <JobCard key={index} job={job} />
            })}
            <p />
            <h3>
              <a name="trainings">Training</a>
            </h3>
            {training.map((job, index) => {
              return <JobCard key={index} job={job} />
            })}
            <p />
            <h3>
              <a name="internships">Internships</a>
            </h3>
            {internships.map((job, index) => {
              return <JobCard key={index} job={job} />
            })}
            <p />
            <h3>
              <a name="certificates">Certificates</a>
            </h3>
            {certificates.map((job, index) => {
              return <JobCard key={index} job={job} />
            })}
            <p />
          </div>
        )}
      </StyledCv>
    )
  }
}

Cv.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }),
}

export default Cv
