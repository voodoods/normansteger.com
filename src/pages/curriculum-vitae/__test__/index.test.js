import React from 'react'
import { shallow } from 'enzyme'
import Cv from '../'

jest.mock('../job-set')

describe('<Cv /> component index', () => {
  it('expects component to match snapshot', () => {
    const props = {
      location: {
        pathname: '/some/path',
      },
      jobSet: [
        {
          group: 'Bla bla',
          title: 'Bla bla',
          company: 'Bla bla',
          description: 'Bla bla',
          from: '05/01/2017',
          to: '05/01/2018',
        },
      ],
      jobGroups: {
        internships: [
          {
            title: 'bla',
            company: 'bla',
            description: 'bla',
            from: 'bla',
            to: 'bla',
          },
        ],
        training: [
          {
            title: 'bla',
            company: 'bla',
            description: 'bla',
            from: 'bla',
            to: 'bla',
          },
        ],
        employments: [
          {
            title: 'bla',
            company: 'bla',
            description: 'bla',
            from: 'bla',
            to: 'bla',
          },
        ],
        certificates: [
          {
            title: 'bla',
            company: 'bla',
            description: 'bla',
            from: 'bla',
            to: 'bla',
          },
        ],
      },
    }
    const wrapper = shallow(<Cv {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
