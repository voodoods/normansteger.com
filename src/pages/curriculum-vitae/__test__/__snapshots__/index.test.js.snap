// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cv /> component index expects component to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withProps(withProps(withProps(Cv)))
    jobGroups={
      Object {
        "certificates": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
        "employments": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
        "internships": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
        "training": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
      }
    }
    jobSet={
      Array [
        Object {
          "company": "Bla bla",
          "description": "Bla bla",
          "from": "05/01/2017",
          "group": "Bla bla",
          "title": "Bla bla",
          "to": "05/01/2018",
        },
      ]
    }
    location={
      Object {
        "pathname": "/some/path",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "jobGroups": Object {
        "certificates": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
        "employments": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
        "internships": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
        "training": Array [
          Object {
            "company": "bla",
            "description": "bla",
            "from": "bla",
            "title": "bla",
            "to": "bla",
          },
        ],
      },
      "jobSet": Array [],
      "location": Object {
        "pathname": "/some/path",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "jobGroups": Object {
          "certificates": Array [
            Object {
              "company": "bla",
              "description": "bla",
              "from": "bla",
              "title": "bla",
              "to": "bla",
            },
          ],
          "employments": Array [
            Object {
              "company": "bla",
              "description": "bla",
              "from": "bla",
              "title": "bla",
              "to": "bla",
            },
          ],
          "internships": Array [
            Object {
              "company": "bla",
              "description": "bla",
              "from": "bla",
              "title": "bla",
              "to": "bla",
            },
          ],
          "training": Array [
            Object {
              "company": "bla",
              "description": "bla",
              "from": "bla",
              "title": "bla",
              "to": "bla",
            },
          ],
        },
        "jobSet": Array [],
        "location": Object {
          "pathname": "/some/path",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
